### PR TITLE
Fix several issues in Kettle

### PR DESCRIPTION
**Pull request by MushuEE**
First, kettle would die if any malformed xml was collected in a job
This is now skipped for failing jobs

Pypy was being installed as pypy3

Also fully reverted cmd change incase it was an issue in writing to BQ
```
root@kettle-57f5b7f9f-h4ph2:/kettle# ln -s /opt/pypy*/bin/pypy3 /usr/bin
root@kettle-57f5b7f9f-h4ph2:/kettle# pypy
bash: pypy: command not found
```
```
root@kettle-57f5b7f9f-h4ph2:/kettle# pypy3
Python 3.6.9 (2ad108f17bdb, Apr 07 2020, 02:59:05)
[PyPy 7.3.1 with GCC 7.3.1 20180303 (Red Hat 7.3.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>>
```
```
root@kettle-57f5b7f9f-h4ph2:/kettle# ls -lart /usr/bin/pypy*
lrwxrwxrwx 1 root root 19 Sep 24 14:25 /usr/bin/pypy -> '/opt/pypy*/bin/pypy'
lrwxrwxrwx 1 root root 37 Sep 25 10:08 /usr/bin/pypy3 -> /opt/pypy3.6-v7.3.1-linux64/bin/pypy3
```

#19359
/area kettle
/cc @RobertKielty @spiffxp

### CODE REVIEW

File: kettle/make_json.py [hunk 1]
```diff
@@ -149,7 +149,13 @@ def set_elapsed(self):
 def parse_junit(xml):
     """Generate failed tests as a series of dicts. Ignore skipped tests."""
     # NOTE: this is modified from gubernator/view_build.py
-    tree = ET.fromstring(xml)
+
+    try:
+        tree = ET.fromstring(xml)
+    except ET.ParseError:
+        print("Malformed xml, skipping")
```
Comment (spiffxp) [inline kettle/make_json.py:156]: For followup: do we get logging of which tests / jobs had malformed XML?

File: kettle/Dockerfile [hunk 2]
```diff
@@ -33,7 +33,7 @@ RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime \
 RUN pip3 install requests google-cloud-pubsub==0.25.0 google-cloud-bigquery==0.27.0 influxdb ruamel.yaml==0.16
 
 RUN curl -fsSL https://downloads.python.org/pypy/pypy3.6-v7.3.1-linux64.tar.bz2 | tar xj -C opt  && \
-    ln -s /opt/pypy*/bin/pypy /usr/bin
+    ln -s /opt/pypy*/bin/pypy3 /usr/bin
```
Comment (spiffxp) [inline kettle/Dockerfile:36]: I couldn't find anything in release notes for 7.1 - 7.3 that would explain this, super weird

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

File: kettle/make_json.py (hunk 1 above)
Comment (MushuEE) [reply kettle/make_json.py:156]: right before hand we log the bucket but within this method we don't have info if it is malformed. 

I will follow up with a PR to make the logging better 👍

File: kettle/Dockerfile (hunk 2 above)
Comment (MushuEE) [reply kettle/Dockerfile:36]: neither could I, and I have tried it in a few env and still only get `pypy3`

**Review by spiffxp — APPROVED** [review]
/lgtm

**Pull request merged**